### PR TITLE
Pass the info around correctly when describing/evaluating

### PR DESCRIPTION
**Pull request by Drup**
Now, you really only need to implement `Graph.pp`. :]
Also, please put back `Key.Set` in the API ...

### CODE REVIEW

**Comment by Drup** [comment]
Hum, it seems I reindented by mistake and I don't have the same ocp-indent options than you do. Can you commit the .ocp-indent that you want please ?

**Comment by Drup** [comment]
ping.

**Comment by samoht** [comment]
> Hum, it seems I reindented by mistake and I don't have the same ocp-indent options than you do. Can you commit the .ocp-indent that you want please ?

I'm using the default parameters, weird. I've just seen this PR, I'll rebase on top of my less optimal changes.

**Comment by Drup** [comment]
> I'm using the default parameters, weird.

Yes, but I don't. :p

**Comment by samoht** [comment]
> Now, you really only need to implement Graph.pp. :]

How do you do this with your changes? I need the info to get project name, libs and packages

**Comment by Drup** [comment]
You conflated printing info and printing the graph. I just made them separate again.

**Comment by samoht** [comment]
well, I just want `mirage describe` to print info about the current project...

File: app/functoria_app.ml [hunk 1]
```diff
@@ -579,14 +573,15 @@ module Make (P: S) = struct
...
-    let eval context t =
-      let info = Config.eval context t in
+    let eval ~partial context t =
+      let info = Config.eval ~partial context t in
       let context = Key.context ~stage:`Configure (Key.deps info) in
       let f map =
-        show_keys map @@ Key.deps info;
-        Key.eval map info @@ map
+        let e = Key.eval map info @@ map in
+        Log.info "@[<v>%a@]" (Info.pp false) (snd e) ;
```
Comment (Drup) [inline app/functoria_app.ml:583]: This prints info about the project (regardless which command is used). You added verbosity info so you can put that in there.

**Comment by Drup** [comment]
I just added unconditional printing of the `Info.t` after evaluation. It's going to be printed with all the command (which actually also improved some printing that were done before for `mirage configure`).

In order not to flood the output for other command, I added a verbosity flag for the `Info.pp` command, but didn't implemented the command line flag, which you did. :p
Either we turn this verbosity up by default for `describe` or we don't, your choice.

Is that clearer ?

**Comment by samoht** [comment]
yep that's clearer thanks. I'll turn on verbosity when the flag is set.

**Comment by Drup** [comment]
Good! Sorry for the monosyllabic explanation before. :p

**Comment by samoht** [comment]
Can I print the info in the `describe` command a bit above? Not sure to fully understand when things are fully evaluated or not. I guess the describe takes the evaluated graph and info as argument.

**Comment by Drup** [comment]
`evaluated` is a pair of the info (including the context) and a graph. I'm not sure I understand why you want exactly ? "above" what ?

**Comment by samoht** [comment]
See https://github.com/samoht/functoria/commit/bd0635ba4923a7ad20155867b52c0f79183cb95c

**Comment by Drup** [comment]
That looks good.

**Comment by Drup** [comment]
This has been merged manually.

**Pull request closed without merging**